### PR TITLE
Xml participant

### PR DESCRIPTION
**Pull request by jparisu**
Requires:

- #371 
- #376 
- https://github.com/eProsima/DDS-Pipe/pull/41

### CODE REVIEW

File: ddsrouter_test/compose/test_cases/dds_rtps/ddsrouter.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+version: v3.0
+
+xml:
+  raw: |
```
Comment (rsanchez15) [inline ddsrouter_test/compose/test_cases/dds_rtps/ddsrouter.yaml:4]: Extend this test with more routers in different domains configured as follows:

- Loading the xml from the default file
- Loading the xml from the environment variable
- Loading the xml from a file

File: docs/rst/appendixes/glossary.rst [hunk 2]
```diff
@@ -18,6 +18,14 @@ Glossary
         Usually, the machines under the router network cannot be accessed from the outside unless a Port is forwarded
         in the router configuration, or if such host has previously started a TCP communication with the message source.
 
+    QoS
+        **Quality of Service**:
+        Collection of attributes and configurations that manage the behavior of an individual DDS entity.
```
Comment (rsanchez15) [inline docs/rst/appendixes/glossary.rst:23]:
```suggestion
        Collection of attributes and settings that configure the behavior of DDS communications.
```

File: docs/rst/notes/forthcoming_version.rst [hunk 3]
```diff
@@ -13,3 +13,7 @@ This release adds new **requirements**:
 This release has the following **major implementation changes**:
 
 * Main functionality has been moved to new repository `DDS Pipe <https://github.com/eProsima/DDS-Pipe>` to reuse it in other projects.
+
+This release includes the following **features**:
+
+* XML Participant
```
Comment (rsanchez15) [inline docs/rst/notes/forthcoming_version.rst:19]: link to docs

File: docs/rst/user_manual/participants/participant.rst [hunk 4]
```diff
@@ -132,6 +127,12 @@ Below is the list with all the available Participant Kinds.
         - Initial Peers DDS DomainParticipant |br|
           for WAN communication.
 
+    *   - :ref:`user_manual_participants_wan`
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/participant.rst:130]: Wrong link

File: docs/rst/user_manual/participants/xml.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+
+Configuration Example
+=====================
+
+Configure a XML Participant that uses Initial Peers to communicate via WAN using TCP.
+It listens for clients in public IP ``82.0.0.1`` in port ``11600`` in ``TCP``.
+
+.. code-block:: yaml
+
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/xml.rst:58]: Add this to the example

```yaml
xml:                                                              # 1
  files:
    - "./xml_configuration.xml"                                   # 2
  raw: |                                                          # 3
    <?xml version="1.0" encoding="UTF-8" ?>
    <profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles" >
        <participant profile_name="custom_participant_configuration">
            <domainId>1</domainId>
            <rtps></rtps>
        </participant>
    </profiles>
```

File: docs/rst/user_manual/participants/xml.rst [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+* To configure the profile, check the following section
+  :ref:`Configuration section <user_manual_configuration_profile>`.
+
+
+Configuration Example
+=====================
+
+Configure a XML Participant that uses Initial Peers to communicate via WAN using TCP.
+It listens for clients in public IP ``82.0.0.1`` in port ``11600`` in ``TCP``.
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/xml.rst:55]: Wrong description

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by rsanchez15 — CHANGES_REQUESTED** [review]
Rebase and resolve conflicts

File: docs/rst/user_manual/participants/xml.rst [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+|fastdds| supports XML to fully configure a DomainParticipant.
+Using XML configuration, user have whole access to the full configuration of a DDS DomainParticipant.
+Check following `documentation <https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/domainparticipant.html>` for further information about configuring a DDS DomainParticipant with XML.
+For further information regarding how to load XML configuration files to |ddsrouter| check following section :ref:`user_manual_configuration_load_xml`.
+
+.. note::
+
+    This kind of Participant is only meant for expert users.
+    Creating an XML configuration is not an easy task, and there is a high probability that messing with internal configurations of DDS makes DDS Router to not work as expected.
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/xml.rst:19]:
```suggestion
    This kind of Participant is meant for advanced users as XML profiles will overwrite the default internal settings of the DDS Router. 
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: ddsrouter_test/compose/test_cases/dds_rtps/ddsrouter.yaml (hunk 1 above)
Comment (jparisu) [reply ddsrouter_test/compose/test_cases/dds_rtps/ddsrouter.yaml:4]: Loading from default file requires a much more extensible change as the file will not be in the same workspade

File: docs/rst/user_manual/participants/xml.rst (hunk 5 above)
Comment (jparisu) [reply docs/rst/user_manual/participants/xml.rst:58]: This is the example of the Participant. The xml part is described in configuration, and it is linked before

**Review by jparisu — COMMENTED** [review]
ToDo from PR https://github.com/eProsima/DDS-Pipe/pull/41

- [x] Users should set ignore_local_endpoints
- [x] RPC support

File: docs/rst/user_manual/participants/xml.rst [hunk 8]
```diff
@@ -28,6 +28,11 @@ Main use case for this Participant is using **DDS Security**, that requires XML
 
     Add link to security documentation
 
+.. warning::
+
+    This Participant kind does not support :term:`RPC`.
+    Thus services and actions of ROS2 will not work correctly.
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/xml.rst:34]:
```suggestion
    Thus services and actions of ROS 2 will not work correctly.
```

File: docs/rst/user_manual/participants/xml.rst [hunk 9]
```diff
@@ -39,12 +44,37 @@ Kind aliases
...
 
-* To configure the profile, check the following section
-  :ref:`Configuration section <user_manual_configuration_profile>`.
+
+.. _user_manual_participants_xml_profiles:
+
+Create a Fast DDS XML Participant profile
+-----------------------------------------
+
+Whole DomainParticipant configuration settings must be configured via XML, |ddsrouter| will not configure any attribute or QoS for it (that is why this participant is not advisable for non expert users).
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/xml.rst:56]: This was already said. 

```suggestion
Whole DomainParticipant configuration settings must be configured via XML, |ddsrouter| will not configure any attribute or QoS for it.
```

File: docs/rst/user_manual/participants/xml.rst [hunk 10]
```diff
@@ -39,12 +44,37 @@ Kind aliases
...
+                <propertiesPolicy>
+                    <properties>
+                        <property>
+                            <name>fastdds.ignore_local_endpoints</name>
+                            <value>true</value>
+                        </property>
+                    </properties>
+                </propertiesPolicy>
+            </rtps>
+        </participant>
```
Comment (rsanchez15) [inline docs/rst/user_manual/participants/xml.rst:77]: I'll add this to a troubleshooting section and put a warning here redirecting to that section.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: docs/rst/user_manual/participants/xml.rst (hunk 10 above)
Comment (jparisu) [reply docs/rst/user_manual/participants/xml.rst:77]: But this is not a bug or a problem, is just a performance issue

Comment (rsanchez15) [reply docs/rst/user_manual/participants/xml.rst:77]: Then replace the numbered list with bullet points (`*`) and indent the code-block to be inside the list item.

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**